PYTHON-BASIC-master/practice/5_additional_topics/task_1.py

```diff
@@ -0,0 +1,106 @@
+import os
+import json
+from typing import Dict, List
+from decimal import Decimal
+import xml.etree.ElementTree as ET
+from xml.dom import minidom
+
+
+def prettify(file_path: str) -> None:
+    content: str = ''
+    with open(file_path, 'r') as file:
+        content = file.read()
+    reparsed = minidom.parseString(content)
+    pretty_content = reparsed.toprettyxml(indent='   ')
+    with open(file_path, 'w') as file:
+        file.write(pretty_content)
+
+
+def weather(path: str, date: str = '2021-09-25', country: str = 'Spain') -> None:
```

Comment:
Function must do one and only one thing, but do it well
(C) clean code

Reply:
Break this function into many small. The main function must be a sequence of call of the downstream functions/

Reply:
You could define separate functions for reading, calculating and performing XML doc.